modules/basegl/processors/imageprocessing/imagelayer.cpp

```diff
@@ -0,0 +1,120 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2018 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/processors/imageprocessing/imagelayer.h>
+
+#include <inviwo/core/datastructures/image/image.h>
+#include <modules/opengl/openglutils.h>
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/texture/textureunit.h>
+#include <modules/opengl/image/imagegl.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo ImageLayer::processorInfo_{
+    "org.inviwo.ImageLayer",  // Class identifier
+    "Image Layer",            // Display name
```

Comment:
Image Layer To Color?

The naming of the processor is a bit ambigous, it can be also be interpreted as being an image layer.
Perhaps the processor should be called ImageLayerToColor? or ImageLayerToColorLayerConverter

Reply:
Think image ImageLayer is fine

Reply:
That is inherent with our design! At some point it was decided that an Image has at least one color layer plus depth and picking at all times. If you consider this, there is no ambiguity.

Reply:
But the processor converts a layer to the color layer. Not sure if we had some conclusion on this but I think that they should be verbs?

Reply:
And/or be descriptive in what it does